### PR TITLE
Switch projects execution environments to Java 8

### PR DESCRIPTION
**Pull request by briandealwis**
Enable use of Java 8 constructs.

Part of #2557

### CODE REVIEW

**Comment by briandealwis** [comment]
Oops, I mistakenly based this off the `photon-m5` branch.

File: README.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ target platform whenever dependencies are updated.
 ### Steps to import into the Eclipse IDE
 
 
-1. Setup JDK 7 in Eclipse
+1. Setup JDK 8 in Eclipse
 
   1. Select `Window/Preferences` (on Mac `Eclipse/Preferences`).
```
Comment (chanseokoh) [inline README.md]: Down below is "Select the folder that contains the JDK 7 installation".

**Comment by briandealwis** [comment]
- The `third_party` code shouldn't be touched
- The `plugins/com.google.cloud.tools.eclipse.ui.test/build.properties` is already fixed

Not sure why the jdt.core.prefs aren't being hammers in the .status bundles!

**Comment by briandealwis** [comment]
Ah! Some of our bundles are parented against the root, not `plugins/`.

**Comment by briandealwis** [comment]
Missed those errors as .status bundles weren't in my working set.

I'm going to remove these `javacSource` and `javacTarget` entries from `build.properties` to drop redundant warnings ("_The 'javacSource' build entry is not required because it matches the default value_").

**Comment by briandealwis** [comment]
PTAL @chanseokoh

**Pull request merged**